pkg/geo/geomfn/segmentize.go

```diff
@@ -51,12 +51,20 @@ func Segmentize(g geo.Geometry, segmentMaxLength float64) (geo.Geometry, error)
 // segment has a length less than or equal to given maximum segment length.
 // Note: List of points does not consist of end point.
 func segmentizeCoords(a geom.Coord, b geom.Coord, maxSegmentLength float64) ([]float64, error) {
+	if len(a) != len(b) {
```

Comment:
can you test these error cases in the unit tests? look at other places we do `errorTestCases`